dms/models/directory.py

```diff
@@ -273,10 +280,13 @@ def _get_parent_categories(self, access_token):
         return directories
 
     def _get_own_root_directories(self):
+        domain = [("is_hidden", "=", False)]
+        model = self.env["dms.directory"]
+        item = model.search(domain, limit=1, order="level_depth asc")
         return (
-            self.env["dms.directory"]
-            .search([("is_hidden", "=", False), ("parent_id", "=", False)])
-            .ids
+            model.search(domain + [("level_depth", "=", item.level_depth)]).ids
```

Comment:
Why do you need the level_depth?

Reply:
This is necessary for an easy search.

Example:
- Documents
- Documents > Code

If the user only has access to Code, we need to list "top" level directories to which the user has access.

Reply:
Yes, but I don't see the relation... This is something dynamic, not static, depending on your security grants.

Reply:
But with the search method the records are already being filtered according to your permissions.

In this case it is not a permissions problem, we just need to list the correct records, nothing else.

Reply:
That approach is not correct, as you can have access to level 0 directory, but then in another tree, to a level 2 directory. Both should be shown as top level directories for that user.

Reply:
Ok, in that case, how do you suggest to do it in an optimal way and with the search method?

Reply:
Wasn't this solved in portal view to see all the possible directories? How was this done in v11 by MuK?

Reply:
Portal view did not exist in Muk v11, I have changed the approach and solved it in a different way, achieving what you indicated.